### PR TITLE
[MOB-9571] refactors evaluateTree

### PR DESCRIPTION
**Pull request by evantk91**
## 🔹 Jira Ticket(s) if any

* [MOB-9572](https://iterable.atlassian.net/browse/MOB-9572)

## ✏️ Description

> This pull request includes refactoring of remaining criteria completion logic.

[MOB-9572]: https://iterable.atlassian.net/browse/MOB-9572?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by evantk91** [comment]
> ## 🔹 Jira Ticket(s) if any
> * [MOB-9572](https://iterable.atlassian.net/browse/MOB-9572)
> 
> ## ✏️ Description
> > This pull request includes refactoring of remaining criteria completion logic.

Excludes evaluateFieldLogic and evaluateSearchQueries. I want to do on a separate PR since these are fairly complex logic.

**Review by Ayyanchira — APPROVED** [review]
Quite a refactor! Made some nit pick comments.

File: iterableapi/src/main/java/com/iterable/iterableapi/util/CriteriaCompletionChecker.java [hunk 1]
```diff
@@ -243,28 +243,16 @@ public boolean evaluateTree(JSONObject node, JSONArray localEventData) {
...
-                    for (int i = 0; i < searchQueries.length(); i++) {
-                        searchQueries.getJSONObject(i).put("isNot", true);
-                        if (evaluateTree(searchQueries.getJSONObject(i), localEventData)) {
-                            return false;
-                        }
-                    }
-                    return true;
+
+                switch (combinator) {
+                    case "And":
```
[the remark this answers is not in the document]
Comment (evantk91) [reply iterableapi/src/main/java/com/iterable/iterableapi/util/CriteriaCompletionChecker.java:248]: Not sure what you mean by this

File: iterableapi/src/main/java/com/iterable/iterableapi/util/CriteriaCompletionChecker.java [hunk 2]
```diff
@@ -308,20 +324,17 @@ private boolean evaluateSearchQueries(JSONObject node, @NonNull JSONArray localE
     }
 
     private boolean evaluateEvent(JSONArray searchQueries, JSONObject eventData, String combinator) throws JSONException {
-        if (combinator.equals("And")) {
-            if (!evaluateFieldLogic(searchQueries, eventData)) {
+        boolean result = evaluateFieldLogic(searchQueries, eventData);
```
[the remark this answers is not in the document]
Comment (evantk91) [reply iterableapi/src/main/java/com/iterable/iterableapi/util/CriteriaCompletionChecker.java:327]: Hows this suggested by Claude :)

File: iterableapi/src/main/java/com/iterable/iterableapi/util/CriteriaCompletionChecker.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (evantk91) [reply iterableapi/src/main/java/com/iterable/iterableapi/util/CriteriaCompletionChecker.java:248]: I see. The cases are matching what is sent from backend. But we should have consistent naming.

File: iterableapi/src/main/java/com/iterable/iterableapi/util/CriteriaCompletionChecker.java [hunk 3]
```diff
@@ -243,28 +243,16 @@ public boolean evaluateTree(JSONObject node, JSONArray localEventData) {
...
+
+                switch (combinator) {
+                    case "And":
+                        return evaluateAnd(searchQueries, localEventData);
+                    case "Or":
+                        return evaluateOr(searchQueries, localEventData);
+                    case "Not":
+                        return evaluateNot(searchQueries, localEventData);
+                    default:
+                        throw new IllegalArgumentException("Unknown combinator: " + combinator);
```
[the remark this answers is not in the document]
Comment (evantk91) [reply iterableapi/src/main/java/com/iterable/iterableapi/util/CriteriaCompletionChecker.java:255]: Good idea. I will incorporate some logging into other refactors.

**Pull request merged**